lib/ansible/modules/network/cloudengine/ce_snmp_target_host.py

```diff
@@ -0,0 +1,947 @@
+        """ Class init """
+
+        # module
+        argument_spec = kwargs["argument_spec"]
+        self.spec = argument_spec
+        self.module =AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+        # module args
+        self.state = self.module.params['state']
+        self.version = self.module.params['version']
+        self.connect_port = self.module.params['connect_port']
+        self.host_name = self.module.params['host_name']
+        self.domain = "snmpUDPDomain"
+        self.address = self.module.params['address']
+        self.notify_type = self.module.params['notify_type']
+        self.vpn_name = self.module.params['vpn_name']
+        self.recv_port = self.module.params['recv_port']
+        self.security_model = self.module.params['security_model']
+        self.security_name = self.module.params['security_name']
+        self.security_name_v3 = self.module.params['security_name_v3']
+        self.security_level = self.module.params['security_level']
+        self.is_public_net = self.module.params['is_public_net']
+        self.interface_name = self.module.params['interface_name']
+
+        # config
+        self.cur_cli_cfg = dict()
+        self.cur_netconf_cfg = dict()
+        self.end_netconf_cfg = dict()
+
+        # state
+        self.changed = False
+        self.updates_cmd = list()
+        self.results = dict()
+        self.proposed = dict()
+        self.existing = dict()
+        self.end_state = dict()
+
+    def netconf_get_config(self, conf_str):
+        """ Get configure by netconf """
+
+        xml_str = get_nc_config(self.module, conf_str)
+
+        return xml_str
+
+    def netconf_set_config(self, conf_str):
+        """ Set configure by netconf """
+
+        xml_str = set_nc_config(self.module, conf_str)
+
+        return xml_str
+
+    def check_cli_args(self):
+        """ Check invalid cli args """
+
+        if self.connect_port:
+            if int(self.connect_port) != 161 and (int(self.connect_port) > 65535 or int(self.connect_port) < 1025):
+                self.module.fail_json(
+                    msg='Error: The value of connect_port %s is out of [161, 1025 - 65535].' % self.connect_port)
+
+    def check_netconf_args(self, result):
+        """ Check invalid netconf args """
+
+        need_cfg = True
+        same_flag = True
+        delete_flag = False
+        result["target_host_info"] = []
+
+        if self.host_name:
+
+            if len(self.host_name) > 32 or len(self.host_name) < 1:
+                self.module.fail_json(
+                    msg='Error: The len of host_name is out of [1 - 32].')
+
+            if self.state == "present":
+                if not self.address or not self.notify_type:
+                    self.module.fail_json(
+                        msg='Error: Please input address notify_type at the same time.')
+
+            if self.security_model and self.security_model in ["v1", "v2c"]:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_if
```